app/features/settings/settings-sidebar.tsx

```diff
@@ -0,0 +1,72 @@
+import { Cog } from 'lucide-react';
+import { AnimatePresence, motion } from 'motion/react';
+import {
+  Sidebar,
+  SidebarProvider,
+  SidebarTrigger,
+} from '~/components/ui/sidebar';
+import {
+  SettingsSidebarProvider,
+  useSettingsSidebar,
+} from './settings-sidebar-provider';
+import { AddAccountView } from './views/add-account-view';
+import { AdvancedSettingsView } from './views/advanced-settings-view';
+import { AllAccountsView } from './views/all-accounts-view';
+import { AppearanceView } from './views/appearance-view';
+import { ContactsView } from './views/contacts-view';
+import { EditProfileView } from './views/edit-profile-view';
+import { MainView } from './views/main-view';
+import { QrView } from './views/qr-view';
+import { SingleAccountView } from './views/single-account-view';
+
+export function SettingsSidebarContent() {
+  const { state } = useSettingsSidebar();
+  const { view } = state;
+
+  return (
+    <Sidebar side="right">
```

Comment:
what is our policy with sidebar state when closed? do we keep the state (selected view) or do we reset everything? right now the view is kept but do we plan to keep it that way?

Reply:
yea I think so, similar to how you made it work in the V0. I'll add that once we decide on a direction for using routing or not